### PR TITLE
Auto expand room list on search

### PR DESCRIPTION
**Pull request by JorikSchellekens**
https://matrix.to/#/!bEWtlqtDwCLFIAKAcv:matrix.org/$9OXotye7LUm2DcZSpDczd0KcB68nLCUvIbCTu1gZ5K8?via=matrix.org&via=chat.weho.st&via=t2l.io

fixes: https://github.com/vector-im/riot-web/issues/14308

relates to:  https://github.com/vector-im/riot-web/issues/13635

### CODE REVIEW

**Comment by t3chguy** [comment]
Wouldn't it be saner for RoomSubList2's to have a forceExpanded prop so the Layout doesn't have to concern itself with filtering?

**Comment by turt2live** [comment]
the layout also can't persist the flag because it'll be all screwed up when the user refreshes.

**Comment by JorikSchellekens** [comment]
So, what's the conclusion here?

**Comment by turt2live** [comment]
probably need a `forceExpand` property instead

File: src/components/views/rooms/RoomSublist2.tsx [hunk 1]
```diff
@@ -47,6 +47,7 @@ import { Direction } from "re-resizable/lib/resizer";
 import { polyfillTouchEvent } from "../../../@types/polyfill";
 import { RoomNotificationStateStore } from "../../../stores/notifications/RoomNotificationStateStore";
 import RoomListLayoutStore from "../../../stores/room-list/RoomListLayoutStore";
+import { Action } from "../../../dispatcher/actions";
```
Comment (turt2live) [inline src/components/views/rooms/RoomSublist2.tsx:50]: shouldn't be needed

File: src/dispatcher/actions.ts [hunk 2]
```diff
@@ -84,4 +84,14 @@ export enum Action {
      * Changes room based on room list order and payload parameters. Should be used with ViewRoomDeltaPayload.
      */
     ViewRoomDelta = "view_room_delta",
+
+    /**
+     * Informs the room list when room filtering has begun. No additional payload information required.
+     */
+    StartRoomFilter = "start_room_filter",
```
Comment (turt2live) [inline src/dispatcher/actions.ts:91]: oh, this also isn't required

File: src/components/views/rooms/RoomList2.tsx [hunk 3]
```diff
@@ -191,7 +191,7 @@ export default class RoomList2 extends React.Component<IProps, IState> {
                     show_room_tile: true, // to make sure the room gets scrolled into view
                 });
             }
-        }
+        }
```
Comment (turt2live) [inline src/components/views/rooms/RoomList2.tsx:194]: space

File: src/dispatcher/actions.ts (hunk 2 above)
Comment (JorikSchellekens) [reply src/dispatcher/actions.ts:91]: and the stop, getting rid of them now

**Pull request merged**